src/terminal/terminal.rs

```diff
@@ -569,10 +569,22 @@ where
     where
         F: FnOnce(&mut Buffer),
     {
-        if !matches!(self.viewport, Viewport::Inline(_)) {
-            return Ok(());
+        match self.viewport {
+            #[cfg(feature = "scrolling-regions")]
+            Viewport::Inline(_) => self.insert_before_scrolling_regions(height, draw_fn),
+            #[cfg(not(feature = "scrolling-regions"))]
+            Viewport::Inline(_) => self.insert_before_no_scrolling_regions(height, draw_fn),
```

Comment:
Given these are mutually exclusive function definitions would it make sense to name these both the same name (insert_before_impl) and only put the config on the functions instead of within it?

Reply:
I originally did that, but then I thought about the doc comment. I'm not sure how conditional compilation works with doc comments. I'll look into it.

Reply:
Ah that makes sense - I'm fine with leaving this as-is.